metric-example/query-by-example.md

```diff
@@ -0,0 +1,516 @@
+
+| Child Dimension Type | Child Synonyms | :parent Dimension Type | :parent Synonyms |
+|---|---|---|---|
+| `SERVICE_METHOD` | Key request | `SERVICE` | — |
+| `APPLICATION_METHOD` | Key user action  | `APPLICATION` | — |
+| `PROCESS_GROUP_INSTANCE` | Process Instance, Process, Instance | `HOST` | — |
+| `DISK` | — | `HOST` | — |
+| `SYNTHETIC_TEST_STEP` | — | `SYNTHETIC_TEST` | — |
+
+We try accessing the descriptor for `builtin:service.keyRequest.errors.fourxx.rate:parents` and observe that we have gained a new dimension `dt.entity.service`. We know the ID of the service we are interested in and put the new dimension to use by referencing it in a `:filter` transformer and see that the result data meets our expectations:
+```
+GET {{base}}/metrics/query?metricSelector=builtin:service.keyRequest.errors.fourxx.rate:parents:filter(eq(dt.entity.service,SERVICE-0123456789ABCDEF))
+Accept: text/csv
+
+metricId,dt.entity.service,dt.entity.service_method,time,value
+"builtin:service.keyRequest.errors.fourxx.rate:parents:filter(eq(dt.entity.service,SERVICE-0123456789ABCDEF))",SERVICE-0123456789ABCDEF,SERVICE_METHOD-0730DE996E8AA425,2019-04-08 00:00:00,0.0
+"builtin:service.keyRequest.errors.fourxx.rate:parents:filter(eq(dt.entity.service,SERVICE-0123456789ABCDEF))",SERVICE-0123456789ABCDEF,SERVICE_METHOD-0730DE996E8AA425,2019-04-15 00:00:00,0.0
+"builtin:service.keyRequest.errors.fourxx.rate:parents:filter(eq(dt.entity.service,SERVICE-0123456789ABCDEF))",SERVICE-0123456789ABCDEF,SERVICE_METHOD-13A5BE527CDA803D,2019-04-08 00:00:00,0.0
+"builtin:service.keyRequest.errors.fourxx.rate:parents:filter(eq(dt.entity.service,SERVICE-0123456789ABCDEF))",SERVICE-0123456789ABCDEF,SERVICE_METHOD-13A5BE527CDA803D,2019-04-15 00:00:00,0.0
+```
+If we want to lose the service dimension again after filtering, we can use a `:merge(dt.entity.service)`. The order of transformations is again important. The filter cannot precede the `:parents`, since the dimension does not exist at that point.
+
+# Scenario 11: Apdex for Users of iOS 6.x
+
+>**Task**
+>We want to filter a secondary dimension (Operating System) by name, rather than by ID.
+
+Most metrics use entities in at least one dimension. The metric for Apdex split by operating system and version has two entity dimensions and one string dimension, namely the application (primary entity) being measured and the operating system (secondary entity) and version (string) for which the data is valid.
+
+When we wish to only obtain data for a specific application, of which we know the name, we may use a scope expression with the `entity` predicate. We cannot use the same technique to instead filter operating systems by name, since scopes do not apply to secondary entities, but we may filter secondary entities by name using a transformer chain. For this, we use a combination of `:name`s and `:filter`.

```

Comment:
```suggestion
When we wish to only obtain data for a specific application, of which we know the name, we may use a scope expression with the `entity` predicate. We cannot use the same technique to instead filter operating systems by name, since scopes do not apply to secondary entities, but we may filter secondary entities by name using a transformer chain. For this, we use a combination of `:names` and `:filter`.

```